ocaml/parsing/jane_syntax.ml

```diff
@@ -508,26 +506,36 @@ module Mode_expr = struct
     let loc = { loc with loc_ghost = true } in
     let txt = List.map Const.ghostify txt in
     { loc; txt }
+end
+
+(** Some mode-related constructs *)
+module Modes = struct
```

Comment:
Would it be better if we merge `module Mode_expr` and `module Modes`? I guess you separated them because the latter uses full-blown jane_syntax but the former doesn't - but that's also not a strong reason?

Reply:
Let's keep them separate. `Mode_expr.t` means something entirely different than `Modes.expression`. I think putting them too close together is confusing.